services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/DownloadDirectoryRequest.java

```diff
@@ -236,8 +251,20 @@ public String toString() {
          */
         Builder delimiter(String delimiter);
 
-        @Override
-        DownloadDirectoryRequest build();
+        /**
+         * Specify a filter that will be used to evaluate which objects should be downloaded from the target directory.
+         * <p>
+         * You can use a filter, for example, to only download objects of a given size, of a given file extension, of a given
+         * last-modified date, etc. See {@link DownloadFilter} for some ready-made implementations. Multiple {@link
+         * DownloadFilter}s can be composed together via the {@code and} and {@code or} methods.
+         * <p>
+         * By default, if no filter is specified, all objects will be downloaded.
+         *
+         * @param filter the filter
+         * @return This builder for method chaining.
+         * @see DownloadFilter
+         */
+        Builder filter(DownloadFilter filter);
```

Comment:
Should we move it to the DownloadDirectoryRequestOverrideConfiguration? The `uploadFileRequestTransformer` is in the `UploadDirectoryOverrideConfiguration`
https://github.com/aws/aws-sdk-java-v2/blob/master/services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/UploadDirectoryRequest.java#L94

Reply:
I suppose `OverrideConfiguration` is intended for parameters that we expect to be configurable at both the client and request level. Do you think it makes sense to have `DownloadFilter` configurable at the client level?

Reply:
`UploadDirectoryOverrideConfiguration` is a request-level configuration though, but yeah, it's a bit confusing right now with this naming. We should definitional revisit it before GA.

Reply:
Understood it's request-level. What I mean is, it seems that the rule of thumb ought to be that anything that can be configured at *both* the client and request level ought to have its request-level option be in an `OverrideConfiguration`.

If we agree on that premise, do we think it would make sense to also make `DownloadFilter` configurable at the client-level, or is it more intuitive as only a request-level option?

Reply:
Gotcha. We don't currently have a way to determine if an option should be part of the `request` itself or the `overrideConfiguration`, but what you said makes sense to me. We just need to make sure it's consistent across the TM. I'll add it to the API surface area review checklist

IMO it's more intuitive as only a request-level option.

Reply:
If we agree it's more intuitive as a request-level-only option then I personally think it would make more sense as a request parameter. But definitely want to be consistent over everything else. Since you've captured it in the surface area review checklist, I'll opt to leave it as is and we can revisit it in a broader context later.